### PR TITLE
Use form.elements instead of custom query

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue #39349 .

### Summary of Changes

Follow the instructions on the issue

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: build/media_source/system/js/fields/validate.es6.js [hunk 1]
```diff
@@ -252,7 +252,7 @@ class JFormValidator {
     const invalid = [];
 
     // Validate form fields
-    const fields = [].slice.call(form.querySelectorAll('input, textarea, select, button, fieldset'));
+    const fields = [].slice.call(form.elements);
```
Comment (wilsonge) [inline build/media_source/system/js/fields/validate.es6.js:255]: Will this get polyfilled in IE when we generate the es5 file?

Comment (dgrammatiko) [reply build/media_source/system/js/fields/validate.es6.js:255]: Nope, but also this should be supported already by IE, it's an old method not something added recently. I think this was introduced in HTML5 (or even earlier), check the `form-associated elements` here (2009): https://www.w3.org/TR/2009/WD-html5-20090212/forms.html

Comment (wilsonge) [reply build/media_source/system/js/fields/validate.es6.js:255]: Your totally right i had the wrong caniuse page 🤦 Ref of correct one: https://caniuse.com/mdn-api_htmlformelement_elements

**Review by wilsonge — APPROVED** [review]

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 75dae9da33407a4ea607ebb569efd75b7a93181a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39350">issues.joomla.org/tracker/joomla-cms/39350</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 75dae9da33407a4ea607ebb569efd75b7a93181a<br /><br />👌<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39350">issues.joomla.org/tracker/joomla-cms/39350</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39350">issues.joomla.org/tracker/joomla-cms/39350</a>.</sub>

**Comment by fancyFranci** [comment]
Nice! Thank you.

**Pull request merged**